Makefile

```diff
@@ -35,8 +35,12 @@ APPLICATION_VERSION = $(PACKAGE_JSON_VERSION)
 S3_BUCKET = resin-production-downloads
 endif
 ifeq ($(RELEASE_TYPE),snapshot)
-CURRENT_COMMIT_HASH = $(shell git log -1 --format="%h")
-APPLICATION_VERSION = $(PACKAGE_JSON_VERSION)+$(CURRENT_COMMIT_HASH)
+CURRENT_COMMIT_INFO = $(shell git log -1 --format="%ct+%h")
+ifneq (,$(findstring -,$(PACKAGE_JSON_VERSION)))
+APPLICATION_VERSION = $(PACKAGE_JSON_VERSION).$(CURRENT_COMMIT_INFO)
+else
+APPLICATION_VERSION = $(PACKAGE_JSON_VERSION)-$(CURRENT_COMMIT_INFO)
+endif
```

Comment:
As this is now reasonably complex, perhaps it ought to be split out into a separate shell script?